lib/modules/manager/sbt/extract.ts

```diff
@@ -267,7 +309,7 @@ const addResolverMatch = q.sym<Ctx>('resolvers').alt(
 
 function registryUrlHandler(ctx: Ctx): Ctx {
```

Comment:
Why don't you apply the fallback to maven central **once** instead of passing it all over? E.g.:

```ts
function registryUrlHandler(ctx: Ctx): Ctx {
  ctx.registryUrls = [...new Set(ctx.registryUrls)];
  if (!ctx.registryUrls.length) {
    ctx.registryUrls.push(REGISTRY_URLS.mavenCentral);
  }

  for (const dep of ctx.deps) {
    dep.registryUrls = [...ctx.registryUrls];
    if (dep.depType === 'plugin') {
      dep.registryUrls.push(SBT_PLUGINS_REPO);
    }
  }
  return ctx;
}
```